src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralService.kt

```diff
@@ -124,13 +127,34 @@ class ReferralService(
     return assignedReferral
   }
 
-  private fun applyOptionalConjunction(existingSpec: Specification<Referral>, predicate: Boolean?, specToJoin: Specification<Referral>): Specification<Referral> {
+  private fun <T> applyOptionalConjunction(existingSpec: Specification<T>, predicate: Boolean?, specToJoin: Specification<T>): Specification<T> {
     if (predicate == null) return existingSpec
     return existingSpec.and(if (predicate) specToJoin else not(specToJoin))
   }
 
+  fun getSentReferralSummaryForUser(user: AuthUser, concluded: Boolean?, cancelled: Boolean?, unassigned: Boolean?, assignedToUserId: String?, page: Pageable?): Iterable<ReferralSummary> {
+    var findSentReferralsSpec = ReferralSpecifications.sent<ReferralSummary>()
```

Comment:
would be good to deduplicate the code here with the code below if possible? perhaps a private parameterized method in this class?

Reply:
Sure Tom, the only reason why I didn't do like that way is because its going to be deleted any ways. I will change it to be consistent

Reply:
creating the specification can be refactored to a common method and deduplicated.

Reply:
awesome - thanks!